### DISCUSSION TITLE
Large size images cause a drastic slowdown after zooming in and out

### DISCUSSION THREAD

**Discussion by bekanui** [fabricjs/fabric.js#8961] [Q&A]
Using an image with very large size (about 13000x30000 in the example below) as the background image cause the canvas to slow down drastically (both zooming and panning) after zooming in enough and then out again.

This behaviour seems to be unrelated to the file size - using a 4MB image, like in the example, or a 25MB image of the same size result in the same outcome.

Strange thing is, the zooming/panning works fine before zooming in, and after it works fine as long as the user doesn't zoom out, at which point it starts to lag a lot. Zooming back in restore the normal performance, but only when staying zoomed in.

Here is a fiddle to reproduce the behaviour; zoom in and out using the wheel and pan by dragging the canvas:
[http://jsfiddle.net/revref/4dgw6knm/72/](http://jsfiddle.net/revref/4dgw6knm/72/)

**Comment by montasellx** [comment]
I have the same problem, have you got any solutions?
